src/extension.ts

```diff
@@ -224,7 +224,7 @@ function buildFile(wsconfig: vscode.WorkspaceConfiguration,
         }
     };
 
-    return invokeCompiler(wsconfig, filename, cfgname).then(diags => {
+    return new Promise<void>(() => {invokeCompiler(wsconfig, filename, cfgname).then(diags => {
```

Comment:
Weird that this new promise is needed since the `.then` on the invokeCompiler promise returns an `Promise<void>`. No need to worry too much about this one though, I'll take a deeper look later :)